portal/__init__.py

```diff
@@ -18,9 +18,54 @@ def create_app(test_config=None):
     from . import db
     db.init_app(app)
 
-    @app.route('/')
+    @app.route('/', methods=["GET", "POST"])
     def index():
-        return render_template('index.html')
+        user_email = None
 
-    return app
+        if request.method == 'POST':
+            email = request.form['email']
+            password = request.form['password']
+
+            print(email)
+            print(password)
+            #GETS DATABASE
+            con = db.get_db()
+            error = None
+            #SETS ERROR TO NONE AT FIRST
+            cur = con.cursor()
+            cur.execute(
+                'SELECT * FROM users WHERE email = %s', (email,)
+            )
+            user = cur.fetchone()
+                #GET THE EMAIL\
+            print(user)
+            if user is None:
+                #FLASH ERROR IS INCORRECT EMAIL
+                error = 'Incorrect email.'
+            elif user[2] != password:
+                error = 'Incorrect password.'
+                #FLASH ERROR IF INC0RRECT PASS
+            if error is None:
+                print('go to session')
+                session.clear()
+                print(user[0])
+                print(user[1])
+                session['user_id'] = user[0]
+                session['user_email'] = user[1]
```

Comment:
instead of setting user info on session cookie, just set user id and use a hook like `@bp.before_app_request` to look at session data and then populate the global `g` object. more info [here](http://flask.pocoo.org/docs/1.0/tutorial/views/)